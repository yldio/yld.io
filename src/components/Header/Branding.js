import React, { useContext } from 'react';
import styled from 'styled-components';
import { StaticQuery, graphql } from 'gatsby';

import LogoLink from './DesktopNav/LogoLink';
import ServiceLink from './DesktopNav/ServiceLink';
import StyledHeaderLink from './DesktopNav/StyledHeaderLink';
import getServiceInfo from '../../utils/getServiceInfo';
import { LogoStyleContext, logoStyleDefaults } from '../../context/PageContext';

const StyledLinksContainer = styled.div`
  flex-grow: 1;
  display: flex;
  align-items: center;
`;

const TopNavBranding = ({ slug }) => (
  <StaticQuery
    query={graphql`
      {
        services: allContentfulService {
          nodes {
            slug
            specialityAreaItems1 {
              id
              slug
              title
            }
            specialityAreaItems2 {
              id
              slug
              title
            }
            specialityAreaItems3 {
              id
              slug
              title
            }
            specialityAreaItems4 {
              id
              slug
              title
            }
          }
        }
        specialities: allContentfulSpeciality {
          nodes {
            slug
            logoColour
          }
        }
      }
    `}
    render={({ services, specialities }) => {
      const { isServicePage, isSpecialityPage, service } = getServiceInfo({
        services,
        specialities,
        slug,
      });

      let {
        fillColorInitial = logoStyleDefaults.fillColorInitial,
        fillColorHover = logoStyleDefaults.fillColorHover,
        textColor = logoStyleDefaults.textColor,
        serviceColor = fillColorInitial,
      } = {
        ...logoStyleDefaults,
        // eslint-disable-next-line react-hooks/rules-of-hooks
        ...useContext(LogoStyleContext),
      };

      return (
        <StyledLinksContainer>
          <LogoLink
            fillColorInitial={fillColorInitial}
            fillColorHover={fillColorHover}
            textColor={textColor}
          />
          {isSpecialityPage || isServicePage ? (
            <ServiceLink
              isSpecialityPage={isSpecialityPage}
              isServicePage={isServicePage}
              service={service}
              color={serviceColor}
            />
          ) : (
            <StyledHeaderLink color={fillColorInitial} as="h3">
              {slug}
            </StyledHeaderLink>
          )}
        </StyledLinksContainer>
      );
    }}
  />
);

export default TopNavBranding;
