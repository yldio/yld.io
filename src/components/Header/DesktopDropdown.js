import React, { PureComponent } from 'react'
import styled from 'styled-components'
import is from 'styled-is'
import remcalc from 'remcalc'
import Chevron from '../Common/Chevron'
import { outlineStyles } from './elements'
import Anchor from '../Common/Anchor'
import theme from '../../utils/theme'
import { NavbarItem } from './Navbar'

const dropDownItemPadding = `padding: 10px 15px 14px 15px;`

const DesktopMenuDropdownContainer = styled(NavbarItem)`
  position: relative;
  transition: color ${props => props.theme.animations.fast} ease-in-out,
    background ${props => props.theme.animations.fast} ease-in-out;
  cursor: pointer;
  background: transparent;

  &:hover {
    background: ${props => props.theme.colors.greyBG};
  }

  ${is('expanded')`
    color: ${props => props.theme.colors.white};
    background: ${props => props.theme.colors.text};

    &:hover {
      background: ${props => props.theme.colors.text};
    }
  `}

  ${is('darkTheme')`
    color: ${props => props.theme.colors.white};

    &:hover {
      background: #3A3553;
    }

    ${is('expanded')`
      background: ${props => props.theme.colors.vibrant};
      color: ${props => props.theme.colors.text};

      &:hover {
        background: ${props => props.theme.colors.vibrant};
        color: #007F56;
      }
    `}
  `}
`

const DesktopMenuDropdownItems = styled.ul`
  position: absolute;
  width: ${remcalc(160)};
  display: flex;
  flex-direction: column;
  top: ${remcalc(48)};
  left: -9999px;
  opacity: 0;
  transition: opacity ${props => props.theme.animations.normal} ease;
  background: ${props => props.theme.colors.greyBG};

  ${is('expanded')`
    left: 0;
    opacity: 1;
  `}

  li {
    display: flex;
  }

  a {
    color: ${props => props.theme.colors.text};
    width: 100%;
    ${dropDownItemPadding}

    &:active,
    &:focus,
    &:hover {
      background: ${props => props.theme.colors.greyBG};
      color: ${props => props.theme.colors.textLight};
    }
    ${outlineStyles}
  }
`

const DropdownNameWrapper = styled.span`
  display: flex;
  align-items: center;
  transition: outline ${props => props.theme.animations.normal} ease-out;
  /* bumping the z-index so that the outline doesn't get behind the dropdown items list */
  z-index: 2;
  ${dropDownItemPadding}
  ${outlineStyles}

  ${is('darkTheme')`
    &:focus {
      outline-color: white;
    } 
  `}
`

const DropdownName = styled.span`
  margin-right: ${remcalc(6)};
`

export default class DesktopMenuDropdown extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      isExpanded: false
    }
    this.ref = React.createRef()
  }

  render() {
    const { items, themeVariation, children } = this.props
    const { isExpanded } = this.state

    return (
      <DesktopMenuDropdownContainer
        aria-haspopup="true"
        expanded={isExpanded}
        aria-expanded={isExpanded}
        onMouseDown={this.toggle}
        onFocus={this.handleFocus}
        onBlur={this.handleBlur}
        darkTheme={themeVariation === theme.variations.dark}
        ref={this.ref}
      >
        <DropdownNameWrapper
          tabIndex="0"
          darkTheme={themeVariation === theme.variations.dark}
        >
          <DropdownName>{children}</DropdownName>
          <Chevron direction={isExpanded ? 'up' : 'down'} />
        </DropdownNameWrapper>
        <DesktopMenuDropdownItems expanded={isExpanded}>
          {items.map(({ to, href, label }, idx) => (
            <li key={idx}>
              <Anchor to={to} href={href} onClick={this.handleItemClick}>
                <span>{label}</span>
              </Anchor>
            </li>
          ))}
        </DesktopMenuDropdownItems>
      </DesktopMenuDropdownContainer>
    )
  }

  toggle = e => {
    this.setState({ isExpanded: !this.state.isExpanded })
  }

  handleItemClick = () => {
    this.setState({ isExpanded: false })
  }

  handleFocus = (e, ...rest) => {
    this.setState({ isExpanded: true })
  }

  handleBlur = () => {
    this.setState({ isExpanded: false })
  }
}
