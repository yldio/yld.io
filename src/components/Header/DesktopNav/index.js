import React from 'react'
import styled from 'styled-components'
import breakpoint from 'styled-components-breakpoint'
import remcalc from 'remcalc'
import OuterAnchorItem from './OuterAnchorItem'
import Dropdown from './Dropdown'
import links from '../utils/navLinks'
import generate from 'shortid'

const TopNavList = styled.ul`
  ${breakpoint('smallPhone')`
    display: none;
  `}

  @media screen and (min-width: 960px) {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: ${remcalc(20)} ${remcalc(0)} ${remcalc(16)};
  }
`

const TopNav = ({ themeVariation }) => (
  <TopNavList>
    {links.map(({ label, dropdownItems, attributes, to, href }) =>
      dropdownItems && dropdownItems.length > 0 ? (
        <Dropdown
          key={generate()}
          themeVariation={themeVariation}
          items={dropdownItems}
          dataEvent={
            attributes && attributes.dataEvent ? attributes.dataEvent : null
          }
        >
          {label}
        </Dropdown>
      ) : (
        <OuterAnchorItem
          key={generate()}
          themeVariation={themeVariation}
          activeClassName="active"
          to={to}
          href={href}
          title={label}
          attributes={attributes}
        >
          {label}
        </OuterAnchorItem>
      )
    )}
  </TopNavList>
)

export default TopNav
