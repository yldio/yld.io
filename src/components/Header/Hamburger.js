import React from 'react'
import styled from 'styled-components'
import breakpoint from 'styled-components-breakpoint'
import remcalc from 'remcalc'
import menu from '../../images/menu.svg'

const HamburgerButton = styled.button`
  display: flex;
  cursor: pointer;
  margin: 0;
  border: none;
  background: transparent;
  padding: ${remcalc(8)} ${remcalc(6)};

  &:focus {
    background: transparent;
    outline: ${remcalc(4)} solid ${props => props.theme.colors.outline};
    color: ${props => props.theme.colors.text};
  }

  ${breakpoint('tablet')`
    display: none;
  `};
`

const Hamburger = ({ onClick }) => (
  <HamburgerButton onClick={onClick}>
    <img src={menu} alt="open menu" />
  </HamburgerButton>
)

export default Hamburger
