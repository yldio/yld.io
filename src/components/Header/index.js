import React, { Component, Fragment } from 'react'
import { Link } from 'gatsby'
import { Row, Col, Grid } from '../grid'
import Flex from 'styled-flex-component'
import styled from 'styled-components'
import { Padding } from 'styled-components-spacing'
import logo from '../../images/logo_animated.gif'
import logoDesign from '../../images/design-logo.svg'
import LogoEngComponent from '../../images/eng-logo-component'
import menu from '../../images/menu.svg'
import close from '../../images/menu_close.svg'
import { MobileMenu, HomeLink, Close, DesktopMenu } from './elements.js'

const HiddenText = styled.h1`
  position: absolute;
  top: -9999px;
  left: -9999px;
`

class Header extends Component {
  state = { menuOpen: false }

  toggleMenu = () => this.setState(state => ({ menuOpen: !state.menuOpen }))

  render() {
    const { location } = this.props

    return (
      <Grid>
        <Row>
          <Col width={[1]}>
            <header>
              <Padding top={2} bottom={3}>
                <Flex alignCenter wrap justifyBetween>
                  <Link to="/">
                    {!location.pathname.includes('engineering') &&
                    !this.props.blue &&
                    !location.pathname.includes('design') ? (
                      <img
                        role="link"
                        tab-index="0"
                        height="48"
                        src={logo}
                        alt="yld logo"
                      />
                    ) : null}

                    {location.pathname.includes('engineering') ||
                    this.props.blue ? (
                      <Fragment>
                        <HiddenText>engineering</HiddenText>
                        <LogoEngComponent
                          boxColour={this.props.logoColour}
                          backgroundBlue={this.props.blue}
                        />
                      </Fragment>
                    ) : null}

                    {this.props.blue ? (
                      <Fragment>
                        <HiddenText>engineering</HiddenText>
                      </Fragment>
                    ) : null}

                    {location.pathname.includes('design') ? (
                      <Fragment>
                        <HiddenText>Design</HiddenText>
                        <img
                          role="link"
                          tab-index="0"
                          height="48"
                          src={logoDesign}
                          alt="yld design logo"
                        />
                      </Fragment>
                    ) : null}
                  </Link>
                  <MobileMenu>
                    <button onClick={this.toggleMenu}>
                      <img src={menu} alt="open menu" />
                    </button>
                  </MobileMenu>
                  <DesktopMenu open={this.state.menuOpen}>
                    <HomeLink right={30}>
                      <Link activeClassName="active" to="/">
                        Home
                      </Link>
                    </HomeLink>
                    <Link reverse activeClassName="active" to="/engineering/">
                      Engineering
                    </Link>
                    <Link activeClassName="active" to="/design/">
                      Design
                    </Link>
                    <Link activeClassName="active" to="/our-work/">
                      Our work
                    </Link>
                    <Link activeClassName="active" to="/contact/">
                      Contact
                    </Link>
                    <a href="https://medium.com/yld-engineering-blog/">Blog</a>
                    <Close onClick={this.toggleMenu}>
                      <img src={close} alt="Close menu" />
                    </Close>
                  </DesktopMenu>
                </Flex>
              </Padding>
            </header>
          </Col>
        </Row>
      </Grid>
    )
  }
}

Header.defaultProps = {
  location: {
    pathname: ''
  }
}

export default Header
