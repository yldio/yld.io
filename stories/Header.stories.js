import React from 'react'
import { storiesOf, addDecorator } from '@storybook/react'
import Theme from './theme'

import Header from '../src/components/Header/'

addDecorator(Theme)

storiesOf('Header', module).add('Header itself', () => <Header path="/" />)
