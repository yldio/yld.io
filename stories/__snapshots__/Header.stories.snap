// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Header Header itself 1`] = `
.c1 {
  max-width: calc(100% - 48px);
  position: relative;
  margin-left: auto;
  margin-right: auto;
  padding-left: 0;
  padding-right: 0;
}

.c3 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
  overflow: hidden;
  padding-left: 0;
  padding-right: 0;
  width: 100%;
  position: initial;
  top: 0;
  left: 0;
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: block;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
  overflow: hidden;
  margin-left: 0;
  margin-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c17 {
  fill: #333333;
}

.c16 {
  border: none;
  background: transparent;
  padding: 0;
  min-width: 3rem;
  min-height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  right: 0;
  top: 0;
  width: 5rem;
  height: 5rem;
  margin: 0.25rem -1.25rem 0.25rem 0.25rem;
  outline: 0.25rem solid transparent;
}

.c16:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c16:active {
  outline: none;
}

.c18 {
  display: none;
}

.c15 {
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
}

.c15:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c15:active {
  outline: none;
}

.c14 {
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fff;
  color: #333333;
}

.c14:hover,
.c14 > a:hover {
  background: #333333;
  color: #a9a9a9;
}

.c14 > a:active {
  background: #333333;
  color: #fff;
}

.c14 > a.active {
  background: #fff;
  color: #828282;
}

.c14 > a.active:active {
  background: #fff;
  color: #828282;
}

.c14 > a.active:hover {
  color: #333333;
}

.c10 {
  stroke: currentColor;
  stroke-width: 0.1875rem;
  fill: none;
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
  -webkit-transition: -webkit-transform 0s ease 0.1s;
  -webkit-transition: transform 0s ease 0.1s;
  transition: transform 0s ease 0.1s;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 {
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
  width: 100%;
  background: #f9f9f9;
  color: #828282;
}

.c13:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c13:active {
  outline: none;
}

.c13:hover,
.c13.active {
  color: #333333;
}

.c7 {
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  position: relative;
  cursor: pointer;
  background: transparent;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c7 > span {
  background: #fff;
  color: #333333;
}

.c7 > span:hover {
  background: #333333;
  color: #a9a9a9;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: outline 300ms ease-out;
  transition: outline 300ms ease-out;
  z-index: 2;
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
}

.c8:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c8:active {
  outline: none;
}

.c9 {
  padding-right: 0.375rem;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11 {
  position: absolute;
  width: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  top: 3rem;
  left: -9999px;
  opacity: 0;
  -webkit-transition: opacity 300ms ease;
  transition: opacity 300ms ease;
  background: #f9f9f9;
  z-index: 999;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 5.25rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: none;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c21 {
  border: none;
  background: transparent;
  padding: 0;
  min-width: 3rem;
  min-height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 5rem;
  height: 5rem;
  margin: 0.25rem;
  outline: 0.25rem solid transparent;
}

.c21:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c21:active {
  outline: none;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.3125rem;
  line-height: 1.5rem;
  padding: 0.5625rem 2.125rem 0.4375rem 1.25rem;
  margin: 0.25rem;
  outline: 0.25rem solid transparent;
  color: #858196;
}

.c24:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c24:active {
  outline: none;
}

.c24:hover,
.c24:focus {
  color: #fff;
}

.c25 {
  max-width: 320px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c23 {
  display: block;
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.3125rem;
  line-height: 1.5rem;
  padding: 0.5625rem 2.125rem 0.4375rem 1.25rem;
  margin: 0.25rem;
  color: #858196;
}

.c23:hover,
.c23:active,
.c23.active {
  color: #fff;
}

.c23:focus {
  color: #fff;
  outline: 0.25rem solid transparent;
}

.c23:focus:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c23:focus:active {
  outline: none;
}

.c22 {
  list-style-type: none;
}

.c19 {
  position: fixed;
  background: #090329;
  height: 100vh;
  width: 100vw;
  left: 0;
  top: 0;
  z-index: 999;
  overflow-y: auto;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

.c0 {
  position: fixed;
  background: #fff;
  width: 100%;
  max-width: unset;
  z-index: 999;
}

@media (min-width:29.4375em) {
  .c1 {
    max-width: calc(100% - 72px);
  }
}

@media (min-width:34.5625em) {
  .c1 {
    max-width: 480px;
  }
}

@media (min-width:43.8125em) {
  .c1 {
    max-width: none;
  }
}

@media (min-width:74.8125em) {
  .c1 {
    max-width: 1100px;
  }
}

@media screen and (min-width:29.4375rem) {
  .c1 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:34.5625rem) {
  .c1 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:43.8125rem) {
  .c1 {
    padding-left: 2.625rem;
    padding-right: 2.625rem;
  }
}

@media screen and (min-width:56.3125rem) {
  .c1 {
    padding-left: 3rem;
    padding-right: 3rem;
  }
}

@media screen and (min-width:74.8125rem) {
  .c1 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:29.4375rem) {
  .c3 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:34.5625rem) {
  .c3 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:43.8125rem) {
  .c3 {
    padding-left: 1.3125rem;
    padding-right: 1.3125rem;
  }
}

@media screen and (min-width:56.3125rem) {
  .c3 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media screen and (min-width:74.8125rem) {
  .c3 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media screen and (min-width:29.4375rem) {
  .c2 {
    margin-left: 0;
    margin-right: 0;
  }
}

@media screen and (min-width:34.5625rem) {
  .c2 {
    margin-left: 0;
    margin-right: 0;
  }
}

@media screen and (min-width:43.8125rem) {
  .c2 {
    margin-left: -1.3125rem;
    margin-right: -1.3125rem;
  }
}

@media screen and (min-width:56.3125rem) {
  .c2 {
    margin-left: -1.5rem;
    margin-right: -1.5rem;
  }
}

@media screen and (min-width:74.8125rem) {
  .c2 {
    margin-left: -1.5rem;
    margin-right: -1.5rem;
  }
}

@media screen and (min-width:960px) {
  .c16 {
    display: none;
  }
}

@media screen and (max-width:959px) {
  .c18 {
    z-index: 9;
    background-color: rgba(51,51,51,0.2);
    content: '';
    display: block;
    position: fixed;
    width: 100vw;
    height: 100vh;
    left: 0;
    top: 0;
    -webkit-transition: opacity 200ms ease-out;
    transition: opacity 200ms ease-out;
    opacity: 0;
    -webkit-transform: translateX(200%);
    -ms-transform: translateX(200%);
    transform: translateX(200%);
  }
}

@media screen and (min-width:960px) {
  .c14 {
    margin-right: 0.375rem;
  }

  .c14:last-child {
    margin-right: 0rem;
  }
}

@media screen and (min-width:960px) {
  .c7 {
    margin-right: 0.375rem;
  }

  .c7:last-child {
    margin-right: 0rem;
  }
}

@media screen and (min-width:960px) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 1.25rem 0rem 1rem;
    padding-right: 0rem;
  }
}

@media screen and (min-width:600px) and (max-width:959px) {
  .c19 {
    width: 18.4375rem;
    left: auto;
    right: 0;
  }
}

@media screen and (min-width:960px) {
  .c19 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      style={
        Object {
          "overflow": "visible",
        }
      }
    >
      <div
        className="c3"
        style={
          Object {
            "overflow": "visible",
          }
        }
        width={
          Array [
            1,
          ]
        }
      >
        <nav
          className="c4"
        >
          <div
            className="c5"
          >
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
              title="Return to Homepage"
            >
              <img
                alt="yld logo"
                height="48"
                role="link"
                src="test-file-stub"
              />
            </a>
          </div>
          <ul
            className="c6"
          >
            <li
              aria-expanded={false}
              aria-haspopup="true"
              className="c7"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="c8"
                tabIndex="0"
              >
                <span
                  className="c9"
                  data-event="navigation-services"
                >
                  Services
                </span>
                <svg
                  className="c10"
                  height="6"
                  viewBox="0 0 6 6"
                  width="6"
                >
                  <polyline
                    points="0,0 6,0 6,6"
                  />
                </svg>
              </span>
              <ul
                className="c11"
              >
                <li
                  className="c12"
                >
                  <a
                    className="c13"
                    href="/engineering/"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    title="Engineering"
                  >
                    Engineering
                  </a>
                </li>
                <li
                  className="c12"
                >
                  <a
                    className="c13"
                    href="/design/"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    title="Design"
                  >
                    Design
                  </a>
                </li>
                <li
                  className="c12"
                >
                  <a
                    className="c13"
                    href="/training/"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    title="Training"
                  >
                    Training
                  </a>
                </li>
              </ul>
            </li>
            <li
              className="c14"
            >
              <a
                className="c15"
                href="/our-work/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                title="Our work"
              >
                Our work
              </a>
            </li>
            <li
              className="c14"
            >
              <a
                className="c15"
                href="/open-source/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                title="Open Source"
              >
                Open Source
              </a>
            </li>
            <li
              className="c14"
            >
              <a
                className="c15"
                data-event="navigation-blog"
                href="/blog/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                title="Blog"
              >
                Blog
              </a>
            </li>
            <li
              aria-expanded={false}
              aria-haspopup="true"
              className="c7"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="c8"
                tabIndex="0"
              >
                <span
                  className="c9"
                  data-event={null}
                >
                  About
                </span>
                <svg
                  className="c10"
                  height="6"
                  viewBox="0 0 6 6"
                  width="6"
                >
                  <polyline
                    points="0,0 6,0 6,6"
                  />
                </svg>
              </span>
              <ul
                className="c11"
              >
                <li
                  className="c12"
                >
                  <a
                    className="c13"
                    href="/about-us/"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    title="Our team"
                  >
                    Our team
                  </a>
                </li>
                <li
                  className="c12"
                >
                  <a
                    className="c13"
                    href="/join-us/"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    title="Join us"
                  >
                    Join us
                  </a>
                </li>
              </ul>
            </li>
            <li
              className="c14"
            >
              <a
                className="c15"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                title="Contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
        <button
          className="c16"
          onClick={[Function]}
        >
          <svg
            className="c17"
            height="18"
            title="open menu"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 16h24v2H0zm0-8h24v2H0zm0-8h24v2H0z"
              fillRule="evenodd"
            />
          </svg>
        </button>
        <div
          className="c18"
          onClick={[Function]}
        />
        <nav
          className="c19"
          open={false}
        >
          <div
            className="c20"
          >
            <button
              className="c21"
              onClick={[Function]}
            >
              <img
                alt="Close menu"
                src="test-file-stub"
              />
            </button>
          </div>
          <ul>
            <li
              className="c22"
            >
              <a
                aria-current="page"
                className="c23 active"
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                states={
                  Object {
                    "default": Array [
                      "
  color: #858196;
",
                    ],
                    "hoverActive": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  }
                }
                style={Object {}}
                title="Home"
              >
                Home
              </a>
            </li>
            <li
              aria-expanded={false}
              aria-haspopup="true"
            >
              <span
                className="c24"
                data-event="navigation-services"
                onFocus={[Function]}
                onMouseDown={[Function]}
                tabIndex="0"
              >
                <span
                  className="c25"
                >
                  Services
                </span>
                <svg
                  className="c10"
                  height="6"
                  viewBox="0 0 6 6"
                  width="6"
                >
                  <polyline
                    points="0,0 6,0 6,6"
                  />
                </svg>
              </span>
            </li>
            <li
              className="c22"
            >
              <a
                className="c23"
                href="/our-work/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                states={
                  Object {
                    "default": Array [
                      "
  color: #858196;
",
                    ],
                    "hoverActive": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  }
                }
                title="Our work"
              >
                Our work
              </a>
            </li>
            <li
              className="c22"
            >
              <a
                className="c23"
                href="/open-source/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                states={
                  Object {
                    "default": Array [
                      "
  color: #858196;
",
                    ],
                    "hoverActive": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  }
                }
                title="Open Source"
              >
                Open Source
              </a>
            </li>
            <li
              className="c22"
            >
              <a
                className="c23"
                data-event="navigation-blog"
                href="/blog/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                states={
                  Object {
                    "default": Array [
                      "
  color: #858196;
",
                    ],
                    "hoverActive": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  }
                }
                title="Blog"
              >
                Blog
              </a>
            </li>
            <li
              aria-expanded={false}
              aria-haspopup="true"
            >
              <span
                className="c24"
                data-event={null}
                onFocus={[Function]}
                onMouseDown={[Function]}
                tabIndex="0"
              >
                <span
                  className="c25"
                >
                  About
                </span>
                <svg
                  className="c10"
                  height="6"
                  viewBox="0 0 6 6"
                  width="6"
                >
                  <polyline
                    points="0,0 6,0 6,6"
                  />
                </svg>
              </span>
            </li>
            <li
              className="c22"
            >
              <a
                className="c23"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                states={
                  Object {
                    "default": Array [
                      "
  color: #858196;
",
                    ],
                    "hoverActive": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  }
                }
                title="Contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
